tsconfig.json

```diff
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
     /* Basic Options */
-    "target": "es5",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
+    "target": "esnext",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
```

Comment:
any specific reason we target esnext ? might be dangerous down the road if stuff gets compiled to unsupported features

Reply:
yes, asyncronous iterators.

Reply:
ehh, not the biggest fan of targeting versions which probably are not fully supported unless there is a big win ( something showing perf difference between the async iterators and whatever else gets generated otherwise).
What if someone comes and adds some code that gets compiled into something that is not supported lets say in node 10 ? I guess CI would catch that ? eitehr way, preferably to have supported versions, but if you think its a big win in perf sure